packages/doctype-tile/src/tile-doctype.ts

```diff
@@ -65,29 +61,19 @@ export class TileDoctype extends Doctype {
      * @param context - Ceramic context
      */
     static async makeGenesis(params: DocParams, context: Context): Promise<Record<string, any>> {
-        const metadata = params.metadata? params.metadata : { controllers: [] }
-
-        // check for DID and authentication
-        if (!context.did || !context.did.authenticated) {
-            throw new Error('No DID authenticated')
-        }
-
-        let unique: string
-        if (params.deterministic) {
-            unique = '0'
-        } else {
-            // If 'deterministic' is undefined, default to creating document uniquely
-            unique = base64Encode(randomBytes(12))
+        // If 'deterministic' is undefined, default to creating document uniquely
+        const unique = params.deterministic ? '0' : base64Encode(randomBytes(12))
+
+        const metadata = params.metadata ? params.metadata : { controllers: [] }
+        if (!metadata.controllers || metadata.controllers.length === 0) {
+            if (params.content) {
+                metadata.controllers = [context.did?.id]
```

Comment:
if the DID is not defined this will be `[undefined]`, right?

Reply:
If DID is not defined then the `_signDagJWS` function will throw an error.

Reply:
yeah, but that's happening later in the code. I think we should throw an `No controllers specified` error instead.

Reply:
Why does that matter? I think this should throw `No DID Authenticated` error regardless.

Reply:
`metadata.controllers` will be `[undefined]` until the `_signDagJWS` call

Reply:
If `context.did` is undefined `_signDagJWS` will throw before that.

Reply:
Why does that matter?

Reply:
I fail to see how this could ever be a problem?

Reply:
Instead of 
```
metadata.controllers = [context.did?.id]
``` 
we could write

```
metadata.controllers  = context.did? [context.did.id] : []
```

and controllers list will never be `[undefined]`. If someone later tries to do something with `[undefined]` there could be an issue. I'm just saying. If you think that's not an issue than OK.

Reply:
I don't understand the problem you are trying to solve?

Reply:
```
if (params.content) {
    metadata.controllers = [context.did?.id]
} else {
    throw new Error('No controllers specified')
}
```

I'm saying that we can throw an error if the `metadata.controllers = [undefined]` as well.

Reply:
But why? Seems like it just adds more code without solving any problem?

Reply:
If someone needs to validate or do whatever with the controllers list **before** the `_signDagJWS` call there will be an issue.